client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -344,7 +368,7 @@ protected ShuffleBlockInfo createShuffleBlock(int partitionId, WriterBuffer wb)
         compressed.length,
         crc32,
         compressed,
-        partitionToServers.get(partitionId),
+        partitionAssignmentRetrieveFunc.apply(partitionId),
```

Comment:
When the partition id is not in the partition ids that failed to send the block, there will be an id inconsistency problem.

Reply:
partitionAssignmentRetrieveFunc has a whole task level assignment mapping, so this won't happen

Reply:
partitionAssignmentRetrieveFunc is to get the latest server, but if the partition id is not registered in the latest server, then the sending will fail.

Reply:
partitionAssignmentRetrieveFunc is to get the latest server for the block failures, and for those normal blocks, it also will get the normal server. I refactor the original code to make the invoking side unify the assignment getting logic

Reply:
ohh, sorry, I understand